src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface ExecutionEnv {
+  name(): string;
+  envInfo(): string;
+  wrapCommand(command: Command): Command;
+}
+
+// This is a base ExeuctionEnv.
+// To write a class that inherits ExecutionEnvBase, the following needs to be added
+// 1. ExecutionEnv class itself
+// 2. ExecutionEnv Creation function : named `function createExecutionEnv(name: string): ExecutionEnv`
+//    This will return class for Env that extends ExecutionEnvBase
+// 3. ExecutionEnv list get function : named `function getConnectableEnvs(): Command`
+//    This will return Commend that execute on host and return string array of device list,
+//    could split with '\n'.
```

Comment:
Could you explain a bit about this line? 
Line 31 mentioned string array will be returned. You mean each element in string array could contain '\n' and could splits again?

Reply:
Oh, sorry. I write invalid information. I mean `single line, split by '\n'. each token will be device name.`. I will fix this comment.